packages/core/src/metrics/exports.ts

```diff
@@ -90,6 +92,50 @@ function distribution(aggregator: MetricsAggregatorConstructor, name: string, va
   addToMetricsAggregator(aggregator, DISTRIBUTION_METRIC_TYPE, name, ensureNumber(value), data);
 }
 
+/**
+ * Adds a timing metric.
+ * The metric is added as a distribution metric.
+ *
+ * You can either directly capture a numeric `value`, or wrap a callback function in `timing`.
+ * In the latter case, the duration of the callback execution will be captured as a span & a metric.
+ *
+ * @experimental This API is experimental and might have breaking changes in the future.
+ */
+function timing<T = void>(
+  aggregator: MetricsAggregatorConstructor,
+  name: string,
+  value: number | (() => T),
+  unit: DurationUnit = 'second',
+  data?: Omit<MetricData, 'unit'>,
+): T | void {
+  // callback form
+  if (typeof value === 'function') {
+    const startTime = timestampInSeconds();
+    const span = startInactiveSpan({
+      op: 'metrics.timing',
+      name,
+      startTime,
+      onlyIfParent: true,
+    });
+
+    return handleCallbackErrors(
+      () => value(),
+      () => {
+        // no special error handling necessary
+      },
+      () => {
+        const endTime = timestampInSeconds();
+        const timeDiff = endTime - startTime;
+        distribution(aggregator, name, timeDiff, { ...data, unit: 'second' });
```

Comment:
I thought about if we should allow to convert this into other units, but for now decided against this - this would just increase bundle size, and not sure what the value would be in doing that? This means that for now this disregards the unit passed in when used in callback form.

Reply:
I think it is totally fine being in seconds.